src/app/components/pages/EventDetails.tsx

```diff
@@ -52,16 +51,23 @@ import { IsaacContent } from "../content/IsaacContent";
 import { EditContentButton } from "../elements/EditContentButton";
 import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
 import * as L from "leaflet";
-
-function formatDate(date: Date | number) {
-  return dayjs(date).format("YYYYMMDD[T]HHmmss");
-}
+import { Location } from "../../../IsaacApiTypes";
 
 interface EventDetailsProps {
   match: { params: { eventId: string } };
   location: { pathname: string };
 }
 
+export const formatAddress = (location: Location | undefined) => {
+  if (!location || !location.address) return "Unknown Location";
+  const addressLine1 = location.address?.addressLine1 || "";
```

Comment:
Is the question mark needed in this line (and the following ones)? You have already added a clause against an empty address, so the address will never be null or undefined here, right?